batcher/aligned-batcher/src/lib.rs

```diff
@@ -667,61 +519,48 @@ impl Batcher {
     /// Returns true if the message was replaced in the batch, false otherwise
     async fn handle_replacement_message(
         &self,
-        mut batch_state: tokio::sync::MutexGuard<'_, BatchState>,
+        mut batch_state_lock: MutexGuard<'_, BatchState>,
```

Comment:
This lock needs to be a `tokio::sync` one, as it is held across an `await` point. Otherwise, you'll deadlock.